fdbserver/KeyValueStoreSQLite.actor.cpp

```diff
@@ -93,29 +94,44 @@ struct PageChecksumCodec {
 		SumType sum;
 		SumType *pSumInPage = (SumType *)(pData + dataLen);
 
-		// Write sum directly to page or to sum variable based on mode
-		SumType *sumOut = write ? pSumInPage : &sum;
-		sumOut->part1 = pageNumber; //DO NOT CHANGE
-		sumOut->part2 = 0x5ca1ab1e;
-		hashlittle2(pData, dataLen, &sumOut->part1, &sumOut->part2);
-
-		// Verify if not in write mode
-		if(!write && sum != *pSumInPage) {
-			if(!silent)
-				TraceEvent (SevError, "SQLitePageChecksumFailure")
-					.error(checksum_failed())
-					.detail("CodecPageSize", pageSize)
-					.detail("CodecReserveSize", reserveSize)
-					.detail("Filename", filename)
-					.detail("PageNumber", pageNumber)
-					.detail("PageSize", pageLen)
-					.detail("ChecksumInPage", pSumInPage->toString())
-					.detail("ChecksumCalculated", sum.toString());
-
-			return false;
+		if (write) {
+			// Always write a CRC32 checksum for new pages
+			sum.part1 = 0; // Indicates CRC32 is being used
+			sum.part2 = crc32c_append(0xfdbeefdb, static_cast<uint8_t*>(data), dataLen);
+			memcpy(pSumInPage, &sum, sizeof(SumType));
+			return true;
 		}
 
-		return true;
+		if (pSumInPage->part1 == 0) {
+			// part1 being 0 indicates with high probability that a CRC32 checksum
+			// was used, so check that first. If this checksum fails, there is still
+			// some chance the page was written with hashlittle2, so fall back to checking
+			// hashlittle2
+			sum.part1 = 0;
+			sum.part2 = crc32c_append(0xfdbeefdb, static_cast<uint8_t*>(data), dataLen);
+			if (sum == *pSumInPage) return true;
```

Comment:
I think we can just end the if() main block here, removing lines 113-118, and then bring the 'else' block contents out a level instead of duplicating the hashlittle2 verification sequence.  So basically this form:
```
if(looks like CRC32) {
  //Try CRC32, return true if match
}

// Try hashlittle2, return true if match

// Log error, return false.
```